verticapy/vdataframe.py

```diff
@@ -11567,11 +11581,24 @@ def to_csv(
                         tmp_row += [na_rep]
                     else:
                         tmp_row += [str(item)]
-                file.write("\n" + sep.join(tmp_row))
+                csv_file += "\n" + sep.join(tmp_row)
             current_nb_rows_written += limit
             file_id += 1
-            file.close()
-        return self
+            if n_files == 1 and path:
+                file = open(path, "w+")
+                file.write(csv_file)
+                file.close()
+            elif path:
+                file = open("{0}/{1}.csv".format(path, file_id), "w+")
```

Comment:
There is already "" outside of {0}. And path is also a string. It may lead to ""/folder/subfolder"/".  If it works then fine.

Otherwise:

file = open(path+"/{1}.csv".format(path, file_id), "w+")